# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/interfaces/html/nsIMozBrowserFrame.idl
# Commit: 9a255864f75d
# Full Hash: 9a255864f75ddcf4096b6222d016a914f5a43c8a
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2019-05-23 04:41:59
# Regressor Bug: 1525720
# File Overlap Count: 1
# Description:
#   Bug 1525720, part 3 - Remove method to createRemoteFrameLoader from nsIMozBrowserFrame interface. r=farre
#   
#   This appears unused and adds unneeded surface area for these API's to support.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D31431
# ==============================================================================

diff -r 5f1c1b609ec1 -r 9a255864f75d dom/interfaces/html/nsIMozBrowserFrame.idl
--- a/dom/interfaces/html/nsIMozBrowserFrame.idl	Wed Apr 24 23:05:40 2019 -0500
+++ b/dom/interfaces/html/nsIMozBrowserFrame.idl	Wed Apr 24 23:25:56 2019 -0500
@@ -36,30 +36,6 @@
   [infallible] readonly attribute boolean isolated;
 
   /**
-   * Normally, a frame tries to create its frame loader when its src is
-   * modified, or its contentWindow is accessed.
-   *
-   * disallowCreateFrameLoader prevents the frame element from creating its
-   * frame loader (in the same way that not being inside a document prevents the
-   * creation of a frame loader).  allowCreateFrameLoader lifts this restriction.
-   *
-   * These methods are not re-entrant -- it is an error to call
-   * disallowCreateFrameLoader twice without first calling allowFrameLoader.
-   *
-   * It's also an error to call either method if we already have a frame loader.
-   */
-  void disallowCreateFrameLoader();
-  void allowCreateFrameLoader();
-
-  /**
-   * Create a remote (i.e., out-of-process) frame loader attached to the given
-   * remote tab.
-   *
-   * It is an error to call this method if we already have a frame loader.
-   */
-  void createRemoteFrameLoader(in nsIRemoteTab aRemoteTab);
-
-  /**
    * Initialize the API, and add frame message listener that supports API
    * invocations.
    */
